src/Sema.zig

```diff
@@ -16386,18 +16386,11 @@ fn zirAsm(
     } else sema.code.nullTerminatedString(extra.data.asm_source);
 
     if (is_global_assembly) {
-        if (outputs_len != 0) {
-            return sema.fail(block, src, "module-level assembly does not support outputs", .{});
-        }
-        if (inputs_len != 0) {
-            return sema.fail(block, src, "module-level assembly does not support inputs", .{});
-        }
-        if (extra.data.clobbers != .none) {
-            return sema.fail(block, src, "module-level assembly does not support clobbers", .{});
-        }
-        if (is_volatile) {
-            return sema.fail(block, src, "volatile keyword is redundant on module-level assembly", .{});
-        }
+        if (outputs_len != 0 or
+            inputs_len != 0 or
+            extra.data.clobbers != .none or
+            is_volatile) unreachable; // already checked by AstGen
+
```

Comment:
this is an odd spelling of
```suggestion
        assert(outputs_len == 0); // validated by AstGen
        assert(inputs_len == 0); // validated by AstGen
        assert(extra.data.clobbers == .none); // validated by AstGen
        assert(!is_volatile); // validated by AstGen
```